feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go

```diff
@@ -260,32 +216,27 @@ func TestHardwarecards(t *testing.T) {
 			fwVerValidation:       false,
 			rrValidation:          false,
 			operStatus:            "",
-			parent:                "",
-			pType:                 componentType["transceiver"],
+			parent:                oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_UNSET,
+			pType:                 oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_TRANSCEIVER,
 		},
 	}}
 
 	for _, tc := range cases {
 		t.Run(tc.desc, func(t *testing.T) {
-			r, err := regexp.Compile(tc.regexpPattern)
-			if err != nil {
-				t.Fatalf("Cannot compile regular expression: %v", err)
-			}
-			cards := findMatchedComponents(t, dut, r)
+			cards := components.FindComponentsByType(t, dut, tc.cardFields.pType)
```

Comment:
running this call for every component type is expensive and will make the test so slow. Please cach the values in the first run and use it for the subsequent calls. U may copy the code from here https://github.com/openconfig/featureprofiles/pull/1277

Reply:
All for efficiency, thanks @mojiiba